hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestParquetColumnProjection.scala

```diff
@@ -327,7 +328,9 @@ class TestParquetColumnProjection extends SparkClientFunctionalTestHarness with
 
       assertEquals(expectedRecordCount, rows.length)
       if (expectedBytesRead != -1) {
-        assertEquals(expectedBytesRead, bytesRead)
+        // verify within 10% of margin.
+        val ten_perc: Double = expectedBytesRead * 0.1;
+        assertTrue((expectedBytesRead + ten_perc) >= bytesRead && bytesRead >= (expectedBytesRead - ten_perc))
```

Comment:
Ratio is more intuitive: `(abs(expectedBytes - bytesRead) / expectedBytes) < 0.1`